src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -821,9 +819,11 @@ void SolverInterfaceImpl::mapWriteDataFrom(
   PRECICE_VALIDATE_MESH_ID(fromMeshID);
   impl::MeshContext &   context        = _accessor->meshContext(fromMeshID);
   impl::MappingContext &mappingContext = context.fromMappingContext;
-  if (mappingContext.mapping.use_count() == 0) {
-    PRECICE_ERROR("From mesh \"" << context.mesh->getName()
-                                 << "\", there is no mapping defined");
+  if (not mappingContext.mapping) {
+    PRECICE_ERROR("You attempt to \"mapWriteDataFrom\" mesh "
```

Comment:
`this function` could maybe be more specific, or `mapWriteDataFrom` should be clear that is a preCICE-internal function:

```suggestion
    PRECICE_ERROR("You attempt to call \"mapWriteDataFrom\" on mesh "
```

even further we could write:
```suggestion
    PRECICE_ERROR("You attempt to call the preCICE-internal function \"mapWriteDataFrom\" on mesh "
```

As an alternative, maybe we could change the "You" subject with "preCICE", to clearly distinguish internal problems from problems originating from wrong API usage. Otherwise, a user could be quite confused if they see this (which normally they should not).

```suggestion
    PRECICE_ERROR("preCICE tried to call \"mapWriteDataFrom\" on mesh "
```

Reply:
This is not an internal function. It is part of the API and the error indicates wrong usage of the API together with the configuration.